pkg/image/prune/prune.go

```diff
@@ -335,17 +325,41 @@ func addImageStreamsToGraph(g graph.Graph, streams *imageapi.ImageStreamList, li
 				}
 				imageNode := n.(*imagegraph.ImageNode)
 
+				istExcluded := false
+				dockerImageReference := ""
+
+				if algorithm.excludeImageStreamTagPatterns != nil {
+					dockerImageReference = imageapi.DockerImageReference{
+						Namespace: stream.Namespace,
+						Name:      stream.Name,
+						Tag:       tag,
+					}.Exact()
+
+					for _, pattern := range algorithm.excludeImageStreamTagPatterns {
+						if pattern.MatchString(dockerImageReference) {
+							istExcluded = true
+							break
+						}
+					}
+				}
+
 				kind := oldImageRevisionReferenceKind
-				if algorithm.pruneOverSizeLimit {
-					if exceedsLimits(stream, imageNode.Image, limits) {
-						kind = WeakReferencedImageEdgeKind
+
+				if !istExcluded {
+					if algorithm.pruneOverSizeLimit {
+						if exceedsLimits(stream, imageNode.Image, limits) {
+							kind = WeakReferencedImageEdgeKind
+						} else {
+							kind = ReferencedImageEdgeKind
+						}
 					} else {
-						kind = ReferencedImageEdgeKind
+						if i < algorithm.keepTagRevisions {
+							kind = ReferencedImageEdgeKind
+						}
 					}
 				} else {
-					if i < algorithm.keepTagRevisions {
-						kind = ReferencedImageEdgeKind
-					}
+					glog.V(4).Infof("ImageStreamTag %q is excluded by the regular expression", dockerImageReference)
+					kind = ReferencedImageEdgeKind
```

Comment:
Shouldn't it be WeakReferencedImageEdgeKind?

Reply:
@dmage No. https://github.com/legionus/origin/blob/304d2b846cbaecc6b30a5b692d9b343cf2c95660/pkg/image/prune/prune.go#L635

Reply:
OK, I thought excluded=pruned. s/excluded/protected/ or kept.